includes/class-receiver.php

```diff
@@ -524,28 +521,50 @@ public static function check_dupes( $commentdata ) {
 		}
 
 		$fragment = wp_parse_url( $commentdata['target'], PHP_URL_FRAGMENT );
-		if ( ! empty( $fragment ) ) {
-			// Check for the newer meta value before checking for the more traditional location
-			$args = array(
-				'post_id'    => $commentdata['comment_post_ID'],
-				'meta_query' => array(
+		$args     = array(
+			'post_id'    => $commentdata['comment_post_ID'],
+			'meta_query' => array(
+				array(
+					'relation' => 'OR',
+					// This would catch incoming webmentions with the same source URL
 					array(
 						'key'     => 'webmention_source_url',
-						'value'   => $commentdata['comment_author_url'],
+						'value'   => $commentdata['comment_meta']['webmention_source_url'],
+						'compare' => '=',
+					),
+
+					// This should catch incoming webmentions with the same canonical URL for Bridgy
+					array(
+						'key'     => 'url',
+						'value'   => $commentdata['comment_meta']['webmention_source_url'],
+						'compare' => '=',
+					),
+					// check comments sent via salmon are also dupes
+					// or anyone else who can't use comment_author_url as the original link,
+					// but can use a _crossposting_link meta value.
+					// @link https://github.com/pfefferle/wordpress-salmon
+					array(
+						'key'     => '_crossposting_link',
+						'value'   => $commentdata['comment_meta']['webmention_source_url'],
 						'compare' => '=',
 					),
+
+					// This would catch incoming activitypub matches, which uses source_url
 					array(
-						'key'     => 'webmention_target_fragment',
-						'value'   => $fragment,
+						'key'     => 'source_url',
+						'value'   => $commentdata['comment_meta']['webmention_source_url'],
 						'compare' => '=',
 					),
 				),
-			);
-		} else {
-			$args = array(
-				'post_id'    => $commentdata['comment_post_ID'],
-				'meta_key'   => 'webmention_source_url',
-				'meta_value' => $commentdata['comment_author_url'],
+			),
+		);
+
+		if ( ! empty( $fragment ) ) {
+			// Ensure that if there is a
+			$args['meta_query'][0][] = array(
```

Comment:
this looks a bit tricky, if the array order changes for example. can we make this a bit more robust?